### PR TITLE
Create check-rhcos

### PR DESCRIPTION
**Pull request by joepvd**
This is a monitoring job that is intended to be run on a schedule, and
to post to slack an image of a table of the results of rhcos pipelines.

### CODE REVIEW

File: pyartcd/pyartcd/pipelines/report_rhcos.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        ]
+
+        self.arches = [
+            "amd64",
+            "s390x",
+            "ppc64le",
+            "arm64"
+        ]
+
+        self.rhcos_urls = {
```
Comment (locriandev) [inline pyartcd/pyartcd/pipelines/report_rhcos.py:33]: The 3 fields above are read-only constants, could be moved outside the class definition (better if renamed in capital letters)

File: pyartcd/pyartcd/pipelines/report_rhcos.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        for arch in self.arches:
+            self.result[arch] = dict()
+            self.result[arch] = await self.get_data_for_arch(arch)
+
+    async def get_data_for_arch(self, arch):
+        jenkins = Jenkins(self.rhcos_urls[arch])
+
+        a = dict()
+        for version in self.versions:
+            if arch == 'arm64' and int(version.split('.')[-1]) < 9:
```
Comment (locriandev) [inline pyartcd/pyartcd/pipelines/report_rhcos.py:50]: A comment would be appreciated :) even if I can tell "arm is not applicable for OCP < 4.9)

File: pyartcd/pyartcd/pipelines/report_rhcos.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+            try:
+                good_id = pipeline.get_last_good_buildnumber()
+            except NoBuildData:
+                bad_id = -1
+            try:
+                bad_id = pipeline.get_last_failed_buildnumber()
+            except NoBuildData:
+                bad_id = -1
+
+            if complete_id == good_id:
```
Comment (locriandev) [inline pyartcd/pyartcd/pipelines/report_rhcos.py:73]: 'good_id' might be referenced before assignment

File: pyartcd/pyartcd/pipelines/report_rhcos.py [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+        output = """
+        digraph {
+          node [ shape=none fontname=Helvetica ]
+          n [ label = <
+            <table>
+              <tr>
+        """
+
+        header = "<td></td>"
```
Comment (locriandev) [inline pyartcd/pyartcd/pipelines/report_rhcos.py:99]: HTML rendering could be simplified by using a template engine, e.g. [Jinja](https://jinja.palletsprojects.com/en/3.1.x/)

File: pyartcd/pyartcd/pipelines/report_rhcos.py [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+            'ppc64le': 'https://jenkins-rhcos.cloud.p8.psi.redhat.com',
+            'arm64': 'https://jenkins-rhcos.cloud.arm.psi.redhat.com'
+        }
+
+    async def get_data(self):
+        for arch in self.arches:
+            self.result[arch] = dict()
+            self.result[arch] = await self.get_data_for_arch(arch)
+
+    async def get_data_for_arch(self, arch):
```
Comment (locriandev) [inline pyartcd/pyartcd/pipelines/report_rhcos.py:45]: Does this really need to be async? Can't find no signs of 'await'

File: scheduled-jobs/build/check-rhcos/Jenkinsfile [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    buildDiscarder(logRotator(artifactDaysToKeepStr: '', artifactNumToKeepStr: '100', daysToKeepStr: '', numToKeepStr: '100')),
+    disableConcurrentBuilds(),
+    disableResume(),
+] )
+
+node() {
+    checkout scm
+
+    b = build(
+        job: '../aos-cd-builds/build%2Fcheck-bugs',
```
Comment (locriandev) [inline scheduled-jobs/build/check-rhcos/Jenkinsfile:11]: Wrong called job :)

File: jobs/build/check-rhcos/Jenkinsfile [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+                    )
+                ]
+            ]
+        ]
+    )
+
+    // Check for mock build
+    commonlib.checkMock()
+
+    // Check bugs
```
Comment (locriandev) [inline jobs/build/check-rhcos/Jenkinsfile:38]: // Check RHCOS

**Review by locriandev — CHANGES_REQUESTED** [review]

**Comment by sosiouxme** [comment]
Couple thoughts:

1. since you're generating html already, how about passing that back up to the job and making it the job's description? then we can just go through job status pages to see trends over time.
1. probably best to run this daily in early AM, but maybe only spam the channel on Mon/Tue/Wed?

**Pull request merged**